src/uidl-definitions/types/index.ts

```diff
@@ -15,8 +15,16 @@ export interface ProjectUIDL {
   components?: Record<string, ComponentUIDL>
 }
 
+enum AssetType {
```

Comment:
these changes should also be reflected in the schema, not only in the TS interfaces

Reply:
Hey @alexnm i just pushed by making all the changes, please let me know if we go with this style of declaration in ts files so i will go ahead and update then and update all the files in codebase where we are using them.
~~~
export enum AssetType {
  "style",
}

Reply:
@JayaKrishnaNamburu looks great, please go ahead and add the TS enums. you can also check in the codebase where the strings are used and replace them. curious if it has any impact in running the tests, but all the generators should behave the same after this change.

Reply:
ok, will work changing them 👍

Reply:
@alexnm i don't think we can go ahead with this approach, here is the challenge in changing it into enum in ts files.

All Enum's are converted into objects in return so we will be able to using as ``AssetType.style`` in the code base but which helps in eliminating strings, but when we can't using ``AssetType.style`` in our project UIDL schema right, so the generators are breaking 😅  I need your inputs 😅

Reply:
declaring them in union style all the builds and projects are being generated without breaking

Reply:
please go with the union types for now, we'll figure out later if it makes sense to use enums or not. the @babel/types library we are using for manipulating ASTs is also using a similar approach as far as I can see